lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1196 @@
+                "TERM=xterm",
+                "HOSTNAME=",
+                "container=podman"
+            ],
+            "Hostname": "",
+            "Image": "docker.io/library/busybox:latest",
+            "Labels": null,
+            "OpenStdin": false,
+            "StdinOnce": false,
+            "StopSignal": 15,
+            "Tty": false,
+            "User": {
+                "gid": 0,
+                "uid": 0
+            },
+            "Volumes": null,
+            "WorkingDir": "/"
+        },
+        "ConmonPidFile": "...",
+        "Created": "2019-06-17T19:13:09.873858307+03:00",
+        "Dependencies": [],
+        "Driver": "overlay",
+        "EffectiveCaps": [
+            "CAP_CHOWN",
+            ...
+        ],
+        "ExecIDs": [],
+        "ExitCommand": [
+            "/usr/bin/podman",
+            "--root",
+            ...
+        ],
+        "GraphDriver": {
+            ...
+        },
+        "HostConfig": {
+            ...
+        },
+        "HostnamePath": "...",
+        "HostsPath": "...",
+        "ID": "...",
+        "Image": "...",
+        "ImageName": "docker.io/library/busybox:latest",
+        "IsInfra": false,
+        "LogPath": "/tmp/container/mycontainer.json",
+        "MountLabel": "system_u:object_r:container_file_t:s0:c282,c782",
+        "Mounts": [
+            ...
+        ],
+        "Name": "myservice",
+        "Namespace": "",
+        "NetworkSettings": {
+            "Bridge": "",
+            ...
+        },
+        "Path": "sh",
+        "ProcessLabel": "system_u:system_r:container_t:s0:c282,c782",
+        "ResolvConfPath": "...",
+        "RestartCount": 0,
+        "Rootfs": "",
+        "State": {
+            "Dead": false,
+            "Error": "",
+            "ExitCode": 0,
+            "FinishedAt": "2019-06-17T19:13:10.157518963+03:00",
+            "Healthcheck": {
+                "FailingStreak": 0,
+                "Log": null,
+                "Status": ""
+            },
+            "OOMKilled": false,
+            "OciVersion": "1.0.1-dev",
+            "Paused": false,
+            "Pid": 4083,
+            "Restarting": false,
+            "Running": false,
+            "StartedAt": "2019-06-17T19:13:10.152479729+03:00",
+            "Status": "exited"
+        },
+        "StaticDir": "..."
+    },
+            ...
+    }'
+"""
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.podman.common import run_podman_command
+
+
+def construct_command_from_params(action, params):
+    if action == 'start':
+        return ['start', params['name']]
+    if action == 'stop':
+        return ['stop', params['name']]
+    if action == 'delete':
+        return ['rm', '-f', params['name']]
+    if action in ['create', 'run']:
+        cmd = ['run', '--name', params['name']]
+        if params['detach']:
+            cmd += ['--detach']
+        if params.get('etc_hosts'):
```

Comment:
This is unnecessary, just use a fallback value:
```suggestion
        for host_n_ip in params.get('etc_hosts', {}).items():
        	cmd += ['--add-host', ':'.join(host_n_ip)]
```

Reply:
done

Reply:
Looks like if you have key explicitly defined as "None", then fallback doesn't work, you'll still get None. I missed it, so added default valuse {} and [] for all iteration parameters.